File: algebra/bezout/python/bezout.py
def bezout(a, b, x = 0, prev_x = 1, y = 1, prev_y = 0):
	""" Calculate the Bézout's identity of 'a' and 'b' recursively
		Using the extended euclidean algorithm
	"""

	# 'a' has to be greater than 'b'
	if b > a:
		a, b = b, a

	# calculate the remainder of a/b
	remainder = a % b

	# if remainder is 0, stop here : gcd found
	if remainder == 0:
		return b, x, y

	# else, update x and y, and continue
	quotient = a // b
	prev_x, prev_y, x, y = x, y, quotient*x + prev_x, quotient*y + prev_y
	return bezout(b, remainder, x, prev_x, y, prev_y)

File: algebra/bezout/java/Bezout.java
/**
 * Calculate the Bézout's identity of 'a' and 'b' recursively
 * using the extended euclidean algorithm.
 * <p>
 * Based on the Python implementation created by (https://github.com/prudywsh)
 */
public class Bezout {
    public static double[] bezout(long a, long b, long x, long y) {
        // 'b' must be greater than 'b'
        if (b > a) {
            long temp = a;
            a = b;
            b = temp;
        }

        long remainder = a % b;

        // GCD (greatest common divisor) found
        if (remainder == 0) {
            return new double[]{b, x, y};
        }

        long quotient = (int) (a / b);

        long prevX = x;
        long prevY = y;
        x = quotient * x + prevX;
        y = quotient * y + prevY;

        return bezout(b, remainder, x, y);
    }

    public static double[] bezout(long a, long b) {
        return bezout(a, b, 0, 1);
    }
}
